Review user guide (#979)
ansible/receptor: docs/source/developer_guide.rst

Linters
-------

> All code must pass a suite of Go linters. There is a pre-commit yaml file in the Receptor repository that points to the linter suite. It is best practice to install the pre-commit yaml so that the linters run locally on each commit.

Reviewer: ```suggestion
All code must pass a suite of Go linters.
There is a pre-commit yaml file in the Receptor repository that points to the linter suite.
It is best practice to install the pre-commit yaml so that the linters run locally on each commit.
```

Author: changed the wording but separated the lines

Source code
-----------

> The next couple of sections are aimed to orient developers to the Receptor codebase and provide a starting point for understanding how Receptor works.

Reviewer: ```suggestion
The following sections help orient developers to the Receptor code base and provide a starting point for understanding how Receptor works.
```

``RegisterConfigTypeForApp`` tells the cmdline parser that "tcp-peer" is mapped to the ``TCPDialerCfg{}`` structure.

``main()`` in ``receptor.go`` is the entry point for a running Receptor process.

In ``receptor.go`` (modified for clarity):

.. code-block:: go

    cl.ParseAndRun("receptor.conf", []string{"Init", "Prepare", "Run"})

> A Receptor config file has many action items, such as "node", "work-command", and "tcp-peer". ParseAndRun is how each of these items are instantiated when Receptor starts.

Reviewer: ```suggestion
A Receptor config file has many action items, such as "node", "work-command", and "tcp-peer". ``ParseAndRun`` is how each of these items are instantiated when Receptor starts.
```

Reviewer: Should "node", "work-command", and "tcp-peer" be wrapped in backticks instead of double quotes?

Author: done

        return nil
    }

> This gets a new TCP dialer object and passes it to the netceptor AddBackend method, so that it can be processed further. AddBackend will start proper Go routines that periodically dial the address defined in the TCP dialer structure, which will lead to a proper TCP connection to another Receptor node.

Reviewer: ```suggestion
This gets a new TCP dialer object and passes it to the netceptor ``AddBackend`` method, so that it can be processed further.
``AddBackend`` starts Go routines that periodically dial the address defined in the TCP dialer structure, which results in a TCP connection to another Receptor node.
```

> In general, when studying how the start up process works in Receptor, take a look at the Init, Prepare, and Run methods throughout the code, as these are the entry points to running those specific components of Receptor.

Reviewer: ```suggestion
In general, when studying how the start up process works in Receptor, take a look at the ``Init``, ``Prepare``, and ``Run`` methods throughout the code, as these are the entry points to running those specific components of Receptor.
```

The FromService here is not "ping", but rather the ephemeral service that was created from ``ListenPacket("")`` in ping.go on `foo`.

> With ``trace`` enabled in the Receptor configuration, the following log statements show the reply from `bar`,

Reviewer: ```suggestion
With ``trace`` enabled in the Receptor configuration, the following log statements show the reply from ``bar``,
```
